File: beit-patissier-manage/src/models/Tag.ts
import { ProductTag } from "./ProductTag";


export enum TagTypeEnum {
    Season = 'Season',     
    EventType = 'EventType', 
    Style = 'Style',        
    Taste = 'Taste',        
    Kashrut = 'Kashrut',    
    Audience = 'Audience'  
}

export class Tag {
    tagId: number;
    categoryId: number;
    name: string;
    tagType: TagTypeEnum;
    productTags: ProductTag[];

    constructor(
        tagId: number,
        categoryId: number,
        name: string,
        tagType: TagTypeEnum,
        productTags: ProductTag[] = []
    ) {
        this.tagId = tagId;
        this.categoryId = categoryId;
        this.name = name;
        this.tagType = tagType;
        this.productTags = productTags;
    }
}

export const TagTypeEnumHebrew: Record<TagTypeEnum, string> = {
    [TagTypeEnum.Season]: 'עונה',
    [TagTypeEnum.EventType]: 'סוג אירוע',
    [TagTypeEnum.Style]: 'סגנון',
    [TagTypeEnum.Taste]: 'טעם',
    [TagTypeEnum.Kashrut]: 'כשרות',
    [TagTypeEnum.Audience]: 'קהל יעד'
};

File: BeitPatissierServer/Models/Tag.cs
using System.ComponentModel.DataAnnotations;

namespace BeitPatissierServer.Models
{
    public class Tag
    {
        public int TagId { get; set; }
        public int CategoryId { get; set; }
        public string Name { get; set; } // "חורף", "קיץ", "בוקר עובדים", "בר שיקים", "חלבי", "מתוק" וכו'
        public TagTypeEnum TagType { get; set; } // סוג הקטגוריה
        public ICollection<ProductTag> ProductTags { get; set; } //רשימת ארועים מתאימים
    }

    public enum TagTypeEnum
    {
        [Display(Name = "עונה")]
        Season,     // קיץ, חורף
        [Display(Name = "סוג אירוע")]
        EventType,  // בת מצווה, מסיבת חברות, בוקר עובדים
        [Display(Name = "סגנון")]
        Style,      // בר שיקים, מגשי אירוח
        [Display(Name = "טעם")]
        Taste,      // מתוק/מלוח
        [Display(Name = "כשרות")]
        Kashrut,    // חלבי/פרווה
        [Display(Name = "קהל יעד")]
        Audience    // גברים/נשים/נוער/ילדים
    }

}
